### ISSUE TITLE
Build encountered a problem due to "unused import"?

### ISSUE THREAD

**Issue by jpilgrim** [eclipse/n4js#773]
Just got the following error message during build:

```
Errors occurred during the build.
Errors running builder 'Xtext Project Builder' on project 'my.project'.
There should not be any unused import. Unused=org.eclipse.n4js.n4JS.impl.NamespaceImportSpecifierImpl@72a919ef (flaggedUsedInCode: true) (declaredDynamic: false, alias: React)
```

What's that? Here is the stack trace from the error log:

```
java.lang.AssertionError: There should not be any unused import. Unused=org.eclipse.n4js.n4JS.impl.NamespaceImportSpecifierImpl@53a25ce2 (flaggedUsedInCode: true) (declaredDynamic: false, alias: React)
	at org.eclipse.n4js.transpiler.utils.TranspilerDebugUtils.assertionFailure(TranspilerDebugUtils.java:185)
	at org.eclipse.n4js.transpiler.utils.TranspilerDebugUtils.assertTrue(TranspilerDebugUtils.java:153)
	at org.eclipse.n4js.transpiler.TranspilerComponent.assertTrue(TranspilerComponent.java:349)
	at org.eclipse.n4js.transpiler.es.transform.SanitizeImportsTransformation.assertPostConditions(SanitizeImportsTransformation.java:69)
	at org.eclipse.n4js.transpiler.AbstractTranspiler.transform(AbstractTranspiler.java:222)
	at org.eclipse.n4js.transpiler.AbstractTranspiler.transpile(AbstractTranspiler.java:153)
	at org.eclipse.n4js.transpiler.es.EcmaScriptTranspiler.transpile(EcmaScriptTranspiler.java:159)
	at org.eclipse.n4js.transpiler.es.EcmaScriptSubGenerator.internalDoGenerate(EcmaScriptSubGenerator.java:153)
	at org.eclipse.n4js.generator.AbstractSubGenerator.doGenerate(AbstractSubGenerator.java:123)
	at org.eclipse.n4js.generator.N4JSCompositeGenerator.doGenerate(N4JSCompositeGenerator.java:54)
	at org.eclipse.n4js.ui.building.instructions.BuildInstruction.handleChangedContents(BuildInstruction.java:202)
	at org.eclipse.n4js.ui.building.instructions.BuildInstruction.process(BuildInstruction.java:130)
	at org.eclipse.n4js.ui.building.N4JSGenerateImmediatelyBuilderState.updateMarkers(N4JSGenerateImmediatelyBuilderState.java:250)
	at org.eclipse.xtext.builder.clustering.ClusteringBuilderState.doUpdate(ClusteringBuilderState.java:283)
	at org.eclipse.n4js.ui.building.N4JSGenerateImmediatelyBuilderState.doUpdate(N4JSGenerateImmediatelyBuilderState.java:206)
	at org.eclipse.xtext.builder.builderState.AbstractBuilderState.update(AbstractBuilderState.java:116)
	at org.eclipse.xtext.builder.impl.XtextBuilder.doBuild(XtextBuilder.java:287)
	at org.eclipse.n4js.ui.building.N4JSBuildTypeTrackingBuilder.doBuild(N4JSBuildTypeTrackingBuilder.java:76)
	at org.eclipse.xtext.builder.impl.XtextBuilder.incrementalBuild(XtextBuilder.java:267)
	at org.eclipse.n4js.ui.building.N4JSBuildTypeTrackingBuilder.incrementalBuild(N4JSBuildTypeTrackingBuilder.java:53)
	at org.eclipse.xtext.builder.impl.XtextBuilder.build(XtextBuilder.java:161)
	at org.eclipse.core.internal.events.BuildManager$2.run(BuildManager.java:735)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:42)
	at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:206)
	at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:246)
	at org.eclipse.core.internal.events.BuildManager$1.run(BuildManager.java:301)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:42)
	at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:304)
	at org.eclipse.core.internal.events.BuildManager.basicBuildLoop(BuildManager.java:360)
	at org.eclipse.core.internal.events.BuildManager.build(BuildManager.java:383)
	at org.eclipse.core.internal.events.AutoBuildJob.doBuild(AutoBuildJob.java:144)
	at org.eclipse.core.internal.events.AutoBuildJob.run(AutoBuildJob.java:235)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:55)
```

**Comment by yoosiba** [comment]
Transpiler asserts that after creating intermediate model there are no "unused imports" in it, i.e. symbol table entries (originating from import statements) that are not referenced in the intermediate model of the script. 
Normally this doesn't happen even if original code has real unused imports, as those are removed from intermediate model by the transpiler itself.

Error message says `alias: React`. React is a special case, as in certain cases artificial import of `React` has to be added by the transpiler for `N4JSX` files that do not refer to `React` elements directly, but need to refer to it in compiled code.

In short, this suggests  bug in `N4JSX` handling. Concrete code example would help a lot in investigating the issue.
Note that recently there was change that touched JSX infrastructure in the transpiler, validations and imports analysis. See https://github.com/eclipse/n4js/pull/751

**Comment by mor-n4** [comment]
Which code base have you been building? Which branch? Did you have local changes?
